haystack/pipelines/base.py

```diff
@@ -865,12 +969,23 @@ def save_to_yaml(self, path: Path, return_defaults: bool = False):
         Save a YAML configuration for the Pipeline that can be used with `Pipeline.load_from_yaml()`.
 
         :param path: path of the output YAML file.
+        :param return_defaults: whether to output parameters that have the default values.
+        """
+        config = self.get_config(return_defaults=return_defaults)
+        with open(path, "w") as outfile:
+            yaml.dump(config, outfile, default_flow_style=False)
+
+    def get_config(self, return_defaults: bool = False) -> dict:
+        """
+        Returns a configuration for the Pipeline that can be used with `Pipeline.load_from_config()`.
+
         :param return_defaults: whether to output parameters that have the default values.
         """
         nodes = self.graph.nodes
 
-        pipeline_name = self.root_node.lower()
-        pipelines: dict = {pipeline_name: {"name": pipeline_name, "type": "Pipeline", "nodes": []}}
+        root_node_to_pipeline_name = {"query": "query", "file": "indexing"}
```

Comment:
This one could be a named constant for the class probably. Just to highlight it a bit.